mandatory/3-playing-computer.js

```diff
@@ -39,4 +39,16 @@ for (let i = 0; i < 5; ++i) {
     const e = f1(i, a);
     console.log(e);
   }
+  i;
 }
+
+/**
+  * Answers
+  
+  * 1. Ln31 --> b is not defined hence program throws an error
+  * 3. The value of x, a, d and e is printed
+  * 4. f1 is called once
+  * 5. f2 is called once
+  * 6. value is zero
+  * 7. Value is one
+**/
```

Comment:
the loop is running five times,
at 3. the values of f2 and f1 are also printed. remember you've been told to remove the error
at 4 and 5.  f2 will be called thrice for all even numbers and f1 will be called twice for all odd numbers. remember the loop runs five times. f2 is called for only even numbers and f1 is called for only odd numbers 
at 6. so for the first call of f1, the value of a is 8
at 7. but the original value remains 6 

this question was a bit confusing so I spent much time to figure it out. I am not sure if I am entirely correct. think about my suggestions and let me know if you come up with different ideas. you should try to log a few results to prove yourself. 

otherwise, great job